spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -74,6 +73,7 @@
 import org.springframework.util.ObjectUtils;
 import org.springframework.util.StringUtils;
 
+import kotlin.coroutines.Continuation;
```

Comment:
This is not OK because Kotlin is `optional` dependency and non-Kotlin projects are gong to fail with `ClassNotFoundException`.

I wonder why you don't use `KotlinDetector.isSuspendingFunction()`.
Looks like it has exact logic you are trying to achieve:
```
        if (isKotlinType(method.getDeclaringClass())) {
            Class<?>[] types = method.getParameterTypes();
            if (types.length > 0 && "kotlin.coroutines.Continuation".equals(types[types.length - 1].getName())) {
                return true;
            }
        }
```
It might be extracted from the loop of these params for some optimization, but still the same outcome.

Reply:
Oh, That's a good point. I missed about considering non-Kotlin projects.
I'll use the `isSuspendingFunction` as I first suggested.